static/js/components/Editor_test.js

```diff
@@ -0,0 +1,351 @@
+      executeTestTransaction(wrapper)
+      assert(dispatchStub.called)
+    })
+
+    it("should call the onChange handler", () => {
+      const wrapper = renderEditor()
+      executeTestTransaction(wrapper)
+      assert(onChange.calledWith(""))
+    })
+
+    it("should call onChange with serialized md", () => {
+      const mdStub = sandbox
+        .stub(pmMarkdown.defaultMarkdownSerializer, "serialize")
+        .returns("# MARKDOWN!")
+      executeTestTransaction(renderEditor())
+      assert(onChange.calledWith("# MARKDOWN!"))
+      assert(mdStub.called)
+    })
+  })
+
+  describe("menuClickHandlerCreator", () => {
+    it("menuClickHandlerCreator should return a function", () => {
+      assert.isFunction(
+        renderEditor()
+          .instance()
+          .menuClickHandlerCreator(sandbox.stub())
+      )
+    })
+
+    it("should call passed function with correct args, preventDefault, focus", () => {
+      const commandStub = sandbox.stub()
+      const preventDefault = sandbox.stub()
+      const wrapper = renderEditor()
+      const clickHandler = wrapper
+        .instance()
+        .menuClickHandlerCreator(commandStub)
+      clickHandler({ preventDefault })
+      assert(
+        commandStub.calledWith(
+          wrapper.instance().view.state,
+          wrapper.instance().view.dispatch,
+          wrapper.instance().view
+        )
+      )
+      assert(preventDefault.called)
+      assert(focusStub.called)
+    })
+  })
+
+  describe("menu bar", () => {
+    it("should have the things we tell it do", () => {
+      const wrapper = renderEditor()
+      assert.equal(
+        wrapper.find(".button-group").length,
+        menuBarManifest.length + 1
+      )
+      const allButtons = wrapper.find(".menu-button")
+      assert.equal(allButtons.length, R.flatten(menuBarManifest).length + 1)
+      assert.deepEqual(allButtons.map(button => button.text()), [
+        ...R.flatten(menuBarManifest).map(item => item.icon.props.children),
+        "link"
+      ])
+    })
+
+    //
+    ;[true, false].forEach(markIsActive => {
+      it(`should handle link button click if link active == ${String(
+        markIsActive
+      )}`, () => {
+        const wrapper = renderEditor()
+        const isActiveStub = sandbox
+          .stub(wrapper.instance(), "markIsActive")
+          .returns(markIsActive)
+        const openLinkMenuStub = sandbox.stub(
+          wrapper.instance(),
+          "openLinkMenu"
+        )
+        const { doubleStub, toggleMarkStub } = stubToggleMark()
+        wrapper.instance().handleLinkClick()
+        assert(isActiveStub.called)
+
+        if (markIsActive) {
+          assert(toggleMarkStub.calledWith(schema.marks.link))
+          assert(
+            doubleStub.calledWith(
+              wrapper.instance().view.state,
+              wrapper.instance().view.dispatch
+            )
+          )
+        } else {
+          assert(openLinkMenuStub.called)
```

Comment:
I was under the impression that we prefer using sinon to assert various things about stubbed methods (e.g.: `sinon.assert.called(openLinkMenuStub)` since they're more descriptive when they fail

Reply:
sure, I can use that